src/nvim/fileio.c

```diff
@@ -4475,12 +4475,12 @@ bool vim_fgets(char_u *buf, int size, FILE *fp) FUNC_ATTR_NONNULL_ALL
       tbuf[sizeof(tbuf) - 2] = NUL;
       errno = 0;
       retval = fgets((char *)tbuf, sizeof(tbuf), fp);
-      if (retval == NULL && errno != EINTR) {
+      if (retval == NULL && (errno != EINTR || feof(fp))) {
```

Comment:
Nitpick, but I would check `feof()` first since there's no reason to assume `errno` is valid unless an error happened.

Reply:
I did it that way because last time Justin argued that one shouldn't have to think about whether `feof()` could change `errno` again or not. :>